### PR TITLE
Adding CircleCI config for pyclient

### PR DESCRIPTION
**Pull request by danqrodney**

### CODE REVIEW

**Comment by danqrodney** [comment]
Added stuff for setting up a venv, credentials, etc. PTAL @calbach

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+jobs:
+  test:
+     <<: *defaults
+     steps:
+      - checkout
+      - run:
+          command: cat py/requirements.txt py/swagger-requirements.txt > all_requirements.txt
+      - restore_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "all_requirements.txt" }}
```
Comment (calbach) [inline .circleci/config.yml]: nit: why the "1" in "deps1"?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            . venv/bin/activate
+            ./project.rb setup-env
+      - save_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "all_requirements.txt" }}
+          paths:
+            - "venv"
+      - run:
+          name: Run tests
+          command: |
+            . venv/bin/activate
```
Comment (calbach) [inline .circleci/config.yml]: Would the virtualenv not still be active at this point from above?

File: ci/upload_cloud_creds.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Run this script to upload service-account keys for gcloud on Circle CI.
+# Usage: ./upload_cloud_creds.sh [path to all-of-us-workbench-test service account JSON file]
+# Requires CI_TOKEN environment variable containing a valid Circle CI token.
```
Comment (calbach) [inline ci/upload_cloud_creds.sh]: How do I get such a token?

Comment (calbach) [reply ci/upload_cloud_creds.sh]: (please document)

File: .circleci/config.yml (hunk 1 above)
Comment (danqrodney) [reply .circleci/config.yml]: Removed.

File: .circleci/config.yml (hunk 2 above)
Comment (danqrodney) [reply .circleci/config.yml]: No, each command is run in a different shell, so you have to activate the venv here in order for the tests to use it.

File: ci/upload_cloud_creds.sh (hunk 3 above)
Comment (danqrodney) [reply ci/upload_cloud_creds.sh]: Documented.

**Pull request merged**